### PR TITLE
Frontend auth

### PR DESCRIPTION
**Pull request by ipaulaandreea**

### CODE REVIEW

File: frontend/das_haus/src/App.js [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+
+  const router = createBrowserRouter([
+    {
+      path: '/',
+      element: <RootLayout />,
+      children: [
+        { path: '/', index: true, element: <HomePage /> },
+        {path: 'auth/', element: <AuthenticationPage/>, action: authAction, name:"auth"},
+        {path: 'dummy', element: <DummyProtectedPage/>, loader: routeProtectionLoader}
+
```
Comment (tremuradaniel) [inline frontend/das_haus/src/App.js]: remove whitespace here

File: frontend/das_haus/src/App.js [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+
+
+      ]}])
+      return <RouterProvider router={router} />;
```
Comment (tremuradaniel) [inline frontend/das_haus/src/App.js]: remove whitespace here. 
You can leave the return statement on the last line, with a new line before the return.

File: frontend/das_haus/src/components/AuthForm.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    if (mode === "signup") {
+      let data = JSON.stringify({
+        email: userData.email,
+        password: userData.password,
+      });
+
+      let config = {
+        method: "post",
+        maxBodyLength: Infinity,
+        url: "http://localhost:8000/api/registration",
```
Comment (tremuradaniel) [inline frontend/das_haus/src/components/AuthForm.js]: don't leave the value like this. Move `http://localhost:8000` to `.env` file. Think that when for whatever reason this url will change, you will have to go back to these files and change it in every place you wrote it! It is much better to use a '.env' file for this. Then, each time the url will change, the change will be done in only one place - the .env file.

Bonus! These kind of values will almost 100% be different depending on the environment you project will be on - in most cases, you'll have a `.env` file for development, and a `.env.prod`. No need to add a `.env.prod` now, but keep this in mind!

File: frontend/das_haus/src/images/bg-1.jpg
Comment (tremuradaniel) [inline frontend/das_haus/src/images/bg-1.jpg:1]: Remove this. Be more careful with your pushes!

File: frontend/das_haus/src/pages/Dummy.js
Comment (tremuradaniel) [inline frontend/das_haus/src/pages/Dummy.js:1]: Why do you need this file? Please keep your pushes clean!

File: frontend/das_haus/src/pages/Home.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import {Container} from 'react-bootstrap'
+const HomePage = () => {
+    return (
+        <Container>
+        <h1>Homepage</h1>
+        </Container>
+    )
+}
+
+export default HomePage;
```
Comment (tremuradaniel) [inline frontend/das_haus/src/pages/Home.js]: add new line at the end of a file

File: frontend/das_haus/src/pages/Root.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        <>
+          <NavigationBar/>
+          <main >
+          <Outlet/>
+          </main>
+        </>
+      );
+    };
+    
+    export default RootLayout;
```
Comment (tremuradaniel) [inline frontend/das_haus/src/pages/Root.js]: new line needed here

File: frontend/das_haus/src/styles/login-photo.jpg
Comment (tremuradaniel) [inline frontend/das_haus/src/styles/login-photo.jpg:1]: Good job on taking the initiative and making the frontend prettier!

File: frontend/das_haus/src/util/routeProtection.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import {redirect} from "react-router-dom";
+
+export function routeProtectionLoader() {
+  let isAuthenticated = localStorage.getItem("token");
+  if (!isAuthenticated) {
+    return redirect("/auth?mode=login");
+  }
+  else {
+    return null;
+
```
Comment (tremuradaniel) [inline frontend/das_haus/src/util/routeProtection.js]: remove these 2 new lines here

**Review by tremuradaniel — CHANGES_REQUESTED** [review]
Please look over my observations and change the PR accordingly

File: frontend/das_haus/src/util/routeProtection.js (hunk 6 above)
Comment (ipaulaandreea) [reply frontend/das_haus/src/util/routeProtection.js:10]: This is how react router works - please read the documentation

File: frontend/das_haus/src/styles/login-photo.jpg
Comment (ipaulaandreea) [reply frontend/das_haus/src/styles/login-photo.jpg:1]: thanks!

**Review by tremuradaniel — CHANGES_REQUESTED** [review]
you need do address all the obsarvations

**Review by tremuradaniel — APPROVED** [review]
Lgtm. Merge it

**Pull request merged**